biorxiv/journal_tracker/journal_recommendation_experiment.py

```diff
@@ -0,0 +1,354 @@
+golden_set_df.head()
+
+
+# # Load Pubmed Central Document Vectors
+
+# In[5]:
+
+
+pmc_articles_df = (
+    pd.read_csv(
+        "../../pmc/exploratory_data_analysis/output/pubmed_central_journal_paper_map.tsv.xz", 
+        sep="\t"
+    )
+    .query("article_type=='research-article'")
+)
+print(pmc_articles_df.shape)
+pmc_articles_df.head()
+
+
+# In[6]:
+
+
+journals = pmc_articles_df.journal.value_counts()
+print(journals.shape)
+journals
+
+
+# In[7]:
+
+
+# Filter out low count journals
+pmc_articles_df = pmc_articles_df.query(f"journal in {journals[journals > 100].index.tolist()}")
+print(pmc_articles_df.shape)
+pmc_articles_df.head()
+
+
+# In[8]:
+
+
+pmc_embedding_dict = {
+    int(path.stem[-7:-4]):pd.read_csv(
+        str(path), 
+        sep="\t"
+    )
+    for path in Path("../../pmc/word_vector_experiment/output/").rglob("*tsv.xz")
+}
+pmc_embedding_dict[300].head()
+
+
+# In[9]:
+
+
+full_training_dataset = {
+    dim : (
+        pmc_articles_df
+        .query(f"pmcid not in {golden_set_df.pmcid.tolist()}")
+        [["pmcid", "journal"]]
+        .merge(pmc_embedding_dict[dim], left_on="pmcid", right_on="document")
+        .drop("pmcid", axis=1)
+        .set_index("document")
+    )
+    for dim in pmc_embedding_dict
+}
+
+
+# In[10]:
+
+
+subsampled_training_dataset = {
+    dim : (
+        pmc_articles_df
+        .query(f"pmcid not in {golden_set_df.pmcid.tolist()}")
+        [["pmcid", "journal"]]
+        .merge(pmc_embedding_dict[dim], left_on="pmcid", right_on="document")
+        .drop("pmcid", axis=1)
+        .groupby("journal", group_keys=False)
+        .apply(lambda x: x.sample(min(len(x), 100), random_state=100))
+        .set_index("document")
+    )
+    for dim in pmc_embedding_dict
+}
+
+
+# # Train Recommendation System
+
+# In[11]:
+
+
+knn_model = KNeighborsClassifier(n_neighbors=10)
+
+
+# ## Random Journal Prediction
+
+# The central idea here is to answer the question what is the accuracy when journals are recommended at random?
+
+# In[12]:
+
+
+model = DummyClassifier(strategy='uniform')
+
+
+# In[14]:
+
+
+_ = cross_validation(
+    model, subsampled_training_dataset[300], 
+    dummy_evaluate, cv=10, 
+    random_state=100, top_predictions=10
+)
+
+
+# ## KNearestNeighbors Paper by Paper Comparison - Full Dataset
+
+# Assuming I didn't take mega-journal influence into account, what would the initial recommendation accuracy be?
+
+# In[ ]:
+
+
+_ = cross_validation(
+    knn_model, full_training_dataset[300], 
+    knn_evaluate, cv=10,
+    random_state=100
+)
+
+
+# ## KNearestNeighbors Paper by Paper Comparison Subsampled
+
+# The first idea for a classifier is to compare which papers are similar to other papers. Due to the overflow of PLOS One papers I sub-sampled each journal to have only 100 papers for representation. Then trained a KNearestNeighbors to determine how often does the correct journal appear in the top ten neighbors as well as top twenty neighbors.
```

Comment:
Here you're predicting which a set of possible journal to recommend to article A based on which journals similar articles were published in? And the metric you're using is if the actual journal is in the set of possible journals?

Reply:
> Here you're predicting which a set of possible journal to recommend to article A based on which journals similar articles were published in?

Correct. I'm finding articles that are as close to a given article and noting which journal these articles were published in. 

>And the metric you're using is if the actual journal is in the set of possible journals?

Correct. I'm evaluating my model by answer the question: Does the correct journal appear in the top 10 most closest neighbors to my query? If yes then positive and negative otherwise